tests/release/test_fixture_args.py

```diff
@@ -0,0 +1,96 @@
+import os
+import subprocess
+from pathlib import Path
+
+import pytest
+
+from pseudopeople.interface import (
+    generate_american_community_survey,
+    generate_current_population_survey,
+    generate_decennial_census,
+    generate_social_security,
+    generate_taxes_1040,
+    generate_taxes_w2_and_1099,
+    generate_women_infants_and_children,
+)
+from tests.release.conftest import (
+    DEFAULT_ENGINE,
+    DEFAULT_STATE,
+    DEFAULT_YEAR,
+    FULL_USA_FILEPATH,
+    RI_FILEPATH,
+    _parse_dataset_params,
+)
+
+
+@pytest.fixture()
+def check_subprocess_environment() -> None:
+    if "RUNNING_AS_SUBPROCESS" not in os.environ:
+        pytest.skip("Skipping this test because it's not running as a subprocess")
+
+
+EXPECTED_PARAMETERS = {
+    "census": (
+        generate_decennial_census,
+        None,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        DEFAULT_YEAR,
+    ),
+    "acs": (
+        generate_american_community_survey,
+        FULL_USA_FILEPATH,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        DEFAULT_YEAR,
+    ),
+    "ssa": (generate_social_security, None, "dask", DEFAULT_STATE, DEFAULT_YEAR),
+    "cps": (generate_current_population_survey, None, DEFAULT_ENGINE, "RI", DEFAULT_YEAR),
+    "tax_w2_1099": (
+        generate_taxes_w2_and_1099,
+        None,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        2010,
+    ),
+    "wic": (generate_women_infants_and_children, None, DEFAULT_ENGINE, "RI", 2010),
+    "tax_1040": (
+        generate_taxes_1040,
+        RI_FILEPATH,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        DEFAULT_YEAR,
+    ),
+}
+
+
+@pytest.mark.subprocess_test
+@pytest.mark.usefixtures("check_subprocess_environment")
+def test_parsing_fixture_params(request: pytest.FixtureRequest) -> None:
+    output = _parse_dataset_params(request)
+    # we know output will have a string as the first element but can't type this
+    # while specifying the types of the other elements in output
+    dataset_name: str = output[0]  # type: ignore [assignment]
```

Comment:
This seems to be the only place you use the dataset_name (first output of the parser) - do we really need that? Is the point that you wanted to test that the correct function is tied to a given `--dataset`?

Reply:
We use the dataset name tell the test which parameters to expect. It's not the best solution since you can only have one set of parameters per dataset but I figured this isn't a high priority test anyway.

We also end up using dataset_name in a future PR